src/connections/XmlHttpRequestConnection.js

```diff
@@ -54,6 +64,26 @@ define(["structures/Response", "structures/CAPIError"], function (Response, CAPI
             method = "POST";
         }
 
+        if (xhrCallbacks && Object.keys(xhrCallbacks).length) {
+            if (xhrCallbacks.upload && Object.keys(xhrCallbacks.upload).length) {
+                _assignCallback(XHR.upload, 'onloadstart', xhrCallbacks.upload.onloadstart);
```

Comment:
It doesn't change anything. I wanted to have a list of allowed events.
IMHO, it's a good idea to limit the number of allowed events just to avoid situations when a custom event prevents JS REST Client from performing in a predictable manner.